tests/integration-tests/tests/networking/test_cluster_networking.py

```diff
@@ -80,67 +86,216 @@ def test_existing_eip(existing_eip, pcluster_config_reader, clusters_factory):
+    logging.info(f"Bastion: {bastion_instance}")
+    remote_command_executor = RemoteCommandExecutor(cluster, bastion=bastion_instance)
     scheduler_commands = get_scheduler_commands(scheduler, remote_command_executor)
     fsx_fs_id = get_fsx_fs_id(cluster, region)
     assert_fsx_lustre_correctly_mounted(remote_command_executor, fsx_mount_dir, os, region, fsx_fs_id)
     assert_fsx_lustre_correctly_shared(scheduler_commands, remote_command_executor, fsx_mount_dir)
 
 
+@pytest.mark.usefixtures("enable_vpc_endpoints")
+@pytest.mark.usefixtures("instance")
+def test_cluster_in_no_internet_subnet(
+    region,
+    scheduler,
+    pcluster_config_reader,
+    clusters_factory,
+    vpc_stack,
+    s3_bucket_factory,
+    test_datadir,
+    architecture,
+    os,
+    mpi_variants,
+    bastion_instance,
+):
+    """This test creates a cluster in a subnet with no internet, run osu latency and checks that no failures occur."""
+    bucket_name = s3_bucket_factory()
+    bucket = boto3.resource("s3", region_name=region).Bucket(bucket_name)
+    bucket.upload_file(str(test_datadir / "pre_install.sh"), "scripts/pre_install.sh")
+
+    vpc_default_security_group_id = get_default_vpc_security_group(vpc_stack.cfn_outputs["VpcId"], region)
+    cluster_config = pcluster_config_reader(
+        vpc_default_security_group_id=vpc_default_security_group_id, bucket_name=bucket_name, architecture=architecture
+    )
+    cluster = clusters_factory(cluster_config)
+    assert_that(cluster).is_not_none()
+
+    assert_that(len(get_compute_nodes_instance_ids(cluster.cfn_name, region))).is_equal_to(1)
+
+    remote_command_executor = RemoteCommandExecutor(cluster, bastion=bastion_instance)
+    slurm_commands = SlurmCommands(remote_command_executor)
+
+    logging.info("Checking cluster has no Internet access by trying to access google.com")
+    internet_result = remote_command_executor.run_remote_command("curl -I https://google.com", raise_on_error=False)
+    assert_that(internet_result.failed).is_true()
+
+    for mpi_variant in mpi_variants:
+        logging.info(f"Running OSU benchmark {OSU_BENCHMARK_VERSION} for {mpi_variant}")
+        compile_osu(mpi_variant, remote_command_executor)
+        submission_script = render_jinja_template(
+            template_file_path=test_datadir / f"osu_pt2pt_submit_{mpi_variant}.sh",
+            osu_benchmark_version=OSU_BENCHMARK_VERSION,
+        )
+        result = slurm_commands.submit_script(str(submission_script))
+        job_id = slurm_commands.assert_job_submitted(result.stdout)
+        slurm_commands.wait_job_completed(job_id)
+        slurm_commands.assert_job_succeeded(job_id)
+
+    utils.check_pcluster_list_cluster_log_streams(cluster, os)
+    assert_no_errors_in_logs(remote_command_executor, scheduler)
+    # Check compute node is scaled down after scaledown idle time
+    wait_for_num_instances_in_cluster(cluster.cfn_name, region, 1)
```

Comment:
can we split this in 3 different private functions so to improve readability of the main testing function?